lib/experimental/new_embed.dart

```diff
@@ -495,21 +505,29 @@ major browsers, such as Firefox, Edge (dev channel), or Chrome.
       minSize: [100, 100],
     );
 
-    if (gistId.isNotEmpty) {
-      _loadAndShowGist(gistId, analyze: false);
+    if (gistId.isNotEmpty || sampleId.isNotEmpty) {
+      _loadAndShowGist(analyze: false);
     }
 
     // set enabled/disabled state of various buttons
     editorIsBusy = false;
   }
 
-  Future<void> _loadAndShowGist(String id, {bool analyze = true}) async {
+  Future<void> _loadAndShowGist({bool analyze = true}) async {
+    if (gistId.isEmpty && sampleId.isEmpty) {
+      print('Can\'t load a gist when neither id nor sample_id is present.');
```

Comment:
nit: consider wrapping with `"` so that you don't need to escape the quote

Reply:
Revised the text instead. I refuse to give up single quotes. 😄